setup.sh

```diff
@@ -1,7 +1,49 @@
-git clone https://github.com/michaelpri10/WelcomeBot.git
-cd WelcomeBot
-git submodule init
-git subodule update
-sudo pip install beautifulsoup4
-sudo pip install requests --upgrade
-sudo pip install websocket-client --upgrade
+#!/bin/bash
+#
+# WelcomeBot setup
+#
+
+rev="`git rev-list HEAD --max-count=1 --abbrev-commit 2>/dev/null`"
+
+if [ "$#" -gt 1 ] || ( [ "$#" -eq 1 ] && [ "$1" == "--help" ] ) ;
+then
+    echo "Set up WelcomeBot."
+    echo
+    echo "Usage:"
+    echo "  $0 [--help | --no-clone]"
+    echo
+    echo "Options:"
+    echo "  --help      Print this help message and exit"
+    echo "  --no-clone  Do NOT clone a WelcomeBot git repo (default behavior when inside a repo)"
+    exit 1
+fi
+
+hello="Setting up WelcomeBot"
+if [ "$rev" == "" ] ;
+then
+    echo "$hello."
+else
+    echo "$hello (revision: $rev)."
+fi
+
+if [ "$1" == "--no-clone" ] && [ "$rev" == "" ] ;
+then
+    echo "There's nothing to setup. Please obtain a copy of WelcomeBot source code first."
+    exit 1
+elif [ "$1" == "--no-clone" ] || [ "$rev" != "" ];
+then
+    echo "Cloning supressed, assuming `pwd` as a WelcomeBot git repo directory."
+else
+    echo "Cloning WelcomeBot..."
+    git clone https://github.com/michaelpri10/WelcomeBot.git michaelpri10/WelcomeBot
+    cd "`pwd`/michaelpri10/WelcomeBot"
+fi
+
+git submodule init 2>/dev/null
+git submodule update # 2>/dev/null
```

Comment:
I am wondering whether to send to `/dev/null` `git submodule init`'s output or `git submodule update`'s output? Thus which of them should be used for notificating user about installed git modules? I chose `update`, because it stays silent when the submodule is already installed, and only screams on _update_ (cloning).